pkg/tickers/retry.go

```diff
@@ -0,0 +1,94 @@
+package tickers
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+	"github.com/smartcontractkit/ocr2keepers/pkg/util"
+)
+
+const (
+	retryDelay              = 500 * time.Millisecond
+	totalAttempt            = 3
+	attemptsCacheExpiration = 10 * retryDelay // 5 seconds
+)
+
+type retryTicker struct {
+	timeTicker
+	nextRetries     sync.Map // time.Time -> ocr2keepers.UpkeepPayload
+	payloadAttempts *util.Cache[int]
+}
+
+// Retry adds a retryable result to the retryTicker.
+func (rt *retryTicker) Retry(result ocr2keepers.CheckResult) error {
+	payload := result.Payload
+
+	if result.Retryable {
```

Comment:
I wonder is it easier to read this function if we instead do:

```
if !result.Retryable {
  return fmt.Errorf("Payload %s is not retryable. Skipping.\n", payload.ID)
}

// do the retry stuff here
```

That way you get to drop the "else" block?

Reply:
Good callout.  I think `readability` really depends on people's preference and what have here is a very classic case. It is more natural for some people, including myself, to read code like "if true, do something, else do others". On the other hand, some people want to always short circuit early, "if something, return, then blah blah". If you feel strongly on this, I will make the suggested change.